src/anbox/cmds/system_info.cpp

```diff
@@ -94,14 +95,18 @@ class SystemInformation {
  private:
   void collect_os_info() {
     os_info_.snap_based = (getenv("SNAP") != nullptr);
-    if (fs::exists(os_release_path)) {
+    fs::path path = os_release_path;
+    // If we're running from within a snap the best we can do is to
+    // access the hostfs and read the os-release file from there.
+    if (os_info_.snap_based && fs::exists(host_os_release_path))
+        path = host_os_release_path;
+
+    // Double check that there aren't any permission errors when trying
+    // to access the file (e.g. because of snap confinement)
+    if (fs::exists(path)) {
```

Comment:
In that case the default value "n/a" will be used which is assigned in the struct below.